plugin/src/main/java/com/microsoft/java/bs/gradle/plugin/utils/AndroidUtils.java

```diff
@@ -156,8 +163,57 @@ private static GradleSourceSet convertVariantToGradleSourceSet(Project project,
       gradleSourceSet.setDisplayName(displayName);
 
       // module dependencies
-      Set<GradleModuleDependency> moduleDependencies =
-          AndroidDependencyCollector.getModuleDependencies(project, variant);
+      addModuleDependencies(gradleSourceSet, project, variant);
+
+      // source and resource
+      addSourceAndResources(gradleSourceSet, variant, isUnitTest);
+
+      // resource outputs
+      addResourceOutputs(gradleSourceSet, variant, isUnitTest);
+
+      List<String> compilerArgs = new ArrayList<>();
+
+      // generated sources and source outputs
+      addGeneratedSourceAndSourceOutputs(gradleSourceSet, variant, compilerArgs);
+
+      // classpath
+      addClasspath(gradleSourceSet, variant);
+
+      // Archive output dirs (not relevant in case of android build variants)
+      gradleSourceSet.setArchiveOutputFiles(new HashMap<>());
+
+      // has tests
+      addTests(gradleSourceSet, project, variant, sourceSets);
```

Comment:
Why `addTests()` put inside `convertVariantToGradleSourceSet()`? Can we handle them separately?